### ISSUE TITLE
Bastion: Screen Falls Asleep While Using Xbox 360 Wireless Controller

### ISSUE THREAD

**Issue by ghost** [ValveSoftware/steam-for-linux#1693]
Hi everyone. I am running Ubuntu 12.04 64-bit and playing the game with the official wireless Xbox 360 controller (with the USB receiver). As I play the game, the screen fades out and locks me out of my computer if I don't wake it up with my mouse. I COULD simply make my computer never fall asleep, but for those who don't want to do that (I wouldn't mind it, but some people may), this could be an issue.

Aaaannd here are my specifications, just in case:
OS/Distro: Ubuntu Linux 12.04 64-bit
Desktop Environment: Cinnamon 2D
Motherboard: ASRock Z68 Pro3-M
Processor: Intel Core i7-2600 @ 3.8GHz (Hyperthreading Enabled)
Graphics: Nvidia GeForce GTX 560 (1GB VRAM)
Memory: 16GB DDR3 1333MHz RAM

**Comment by My1** [comment]
so while I don't know about screensavers, at least the screen dims and/or turns off feature has a more or less "obvious" way of turning off.

![image](https://user-images.githubusercontent.com/6696524/54048531-608c6380-41da-11e9-978b-0c3b41606e8f.png)

so apparently software can now go for a more straight power management override, 

sorry @PyGuy95 but personally I think for native linux games, those should be going and sending those signals as they know best when to use them while a steam-global way of doing this cant account for everything and may be too broad.

a game for example could be sending those signals while recieving input, but when nothing is happening still lay the computer idle during "gameplay sections" but when you have like a movie inside the game where the user just watches and isnt supposed to do much inputs it can send a constant override during that time, similar to vivaldi only stopping power management in certain scenarios like for example actively playing audio in a youtube video.

**Comment by ribeirobreno** [comment]
> a game for example could be sending those signals while recieving input  

Input should suspend screensavers and some energy saving features by default, no need for the application (game) to send any signals.
This already is handled fine for keyboard and mouse but not for controllers which **are user input**.

In the case of cut scenes and sections **without user input** like the example from Vivaldi (awesome browser, btw) where an audio and probably video is playing then this is the application's job and it should warn the system that no commands from the user are expected.

An argument can be made about games blocking screensavers from start to finish but screensavers should just respect controller input in the same way as keyboard and mouse input.

---

I don't remember seeing this problem in a long time, i'll check if it still exists here. :)

**Comment by My1** [comment]
I agree that controller input should also be handled as activity, but apparently it isnt always, I had this a few times which is especially noticable on laptops because they usually dim the screen relatively quickly.

but regarding why gamepads dont affect energy management, while I dont know the reason I am fairly sure there has to be SOME reason, after at least windows and linux are and/or were affected. maybe (just me theorizing) the system doesnt see that the joystick currently matters or whatever similarly as if when you are in a console-only mode on the OS, I doubt it'll care much about mouse input, similar to that would be the joystick situation.

**Comment by Bumbadawg** [comment]
@kisak-valve it still happens to this day, Steam controller + Sekiro + intense boss fight is not considered as activity from the system, therefore Steam should send some command (dbus or something) to keep the system awoken.

**Comment by codeman101** [comment]
I confirm it happens with weird xbox 360 controller as well.

**Comment by Zamundaaa** [comment]
This is also happening with Steam Controller in The Witcher III on KDE Plasma (5.16.80 / 5.17 beta).

**Comment by kaimast** [comment]
I can observe this too, albeit with a different game. Has anybody filed an upstream bug with GNOME or KDE about this?

**Comment by codeman101** [comment]
> I can observe this too, albeit with a different game. Has anybody filed an upstream bug with GNOME or KDE about this?

You mean the issue is specific to desktop environments? I thought it was something with the support of the game itself. Obviously not to a specific game but something regarding the games that are affected by this. Now I wish I could test that theory by switching my KDE to something like i3 unfortunately plasma doesn't allow you to do that.

**Comment by soredake** [comment]
https://bugs.kde.org/show_bug.cgi?id=328987

**Comment by Dinth** [comment]
Why wouldnt Steam just set systemd-inhibit when running a game - just as its doing and preventing screensaver/computer sleep when idlying without a game running? At the end of the day, nowadays almost any desktop linux distro is using systemd and so is steam - just its inhibit behaviour is literally an opposite of what its supposed to be? https://github.com/ValveSoftware/steam-for-linux/issues/5607

**Comment by Dinth** [comment]
> Anyone struggling with this issue might want to have a look at joystickwake or gamemode.

Gamemode does fix the issue, but it requires modifying command line option with every single game

> FYI, there is a d-bus API to inhibit screen savers on many desktop
All right, but this is a detail how to inhibit the screensaver/sleep. The fact is that Steam already does that, but only when a game is NOT running, and turns inhibit off when the user actually runs a game. 

>IMHO, the correct solution would be for game controllers to be recognized as input devices in the same places where keyboards and mice
That would be nice indeed, but regardless, when a game is running Steam should inhibit sleep anyway. 

> let's keep in mind that Steam also has to maintain compatibility with a variety of different environments,
Again, regardless whats the actual technical solution of implementing inhibit - Steam already implements it. Just in wrong times.

**Comment by My1** [comment]
> Why wouldnt Steam just set systemd-inhibit when running a game - just as its doing and preventing screensaver/computer sleep when idlying without a game running? At the end of the day, nowadays almost any desktop linux distro is using systemd and so is steam - just its inhibit behaviour is literally an opposite of what its supposed to be? #5607

because that would just be what we in germany call the "wood hammer method", way too simple of an approach and can cause its own issues, the screensaver should trigger even if the game is running, especially on oled screens.

the problem is that controller input does not contribute to "input" in terms of screensaver stuff

regarding the "blocking sleep and screenlocking" yes that's exactly what it does, apparently there are multiple inhibit methods or settings or whatever, some only block sleep and stuff (which is great when e.g. steam is downloading stuff as you would want the PC to properly stay on, but the screen to still turn off) and then there's a mode that also blocks screensavers (which e.g. is used by videoplayers etc.

**Comment by Managor** [comment]
This is likely not an issue any more as Bastion blocks screen locking as is seen in Power Management 

![Image](https://github.com/user-attachments/assets/75f84469-db97-4bce-bed6-dedad7edefa1)

**Comment by kisak-valve** [comment]
Closing per the last comment.
